File: python/22.11/383.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        dic = collections.defaultdict(int)
        for i in magazine:
            dic[i] += 1
        
        for i in ransomNote:
            dic[i] -= 1
            if dic[i] == -1:
                return False
        return True

File: typescript/23.08/383.ts
function canConstruct(ransomNote: string, magazine: string): boolean {
    const mem = {}
    for (const x of magazine) {
        if (mem[x]++) {
            mem[x]++
        } else {
            mem[x] = 1
        }
    }
    for (const x of ransomNote) {
        if (mem[x]) {
            mem[x]--
        } else {
            return false
        }
    }

    return true
};
